src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java

```diff
@@ -491,6 +495,10 @@ public void verifyPostCondition(
     if (getSandboxOptions().useHermetic) {
       checkForConcurrentModifications(context);
     }
+    // We cannot leave the cgroups around and delete them only when we delete the sandboxes
+    // because linux has a hard limit of 65535 memory controllers.
+    // Ref. https://github.com/torvalds/linux/blob/58d4e450a490d5f02183f6834c12550ba26d3b47/include/linux/memcontrol.h#L69
+    cgroups.remove(context.getId()).ifPresent(VirtualCGroup::delete);
```

Comment:
That's interesting, I don't think we've encountered this in our internal builds presumably because we've not hit this limit for the number of sandboxed spawns created during a build. (I think this is fine though).